lib/portage/tests/resolver/ResolverPlayground.py

```diff
@@ -268,6 +270,10 @@ def _create_binpkgs(self, binpkgs):
 		# a dict.
 		items = getattr(binpkgs, 'items', None)
 		items = items() if items is not None else binpkgs
+		binpkg_format = self.settings.get("BINPKG_FORMAT", "xpak")
+		if binpkg_format == "gpkg":
+			gpg = GPG(self.settings)
+			gpg.unlock()
```

Comment:
With the gpg-keepalive feature enabled, this leaks a daemon thread into the background for each ResolverPlayground instance. You need to kill that thread when you're done with it.

Reply:
Maybe use a singleton, like the proxy in lib/portage/util/socks5.py.

Reply:
Also, since threads cause contention for python's GIL, it's nicer to use a process like lib/portage/util/socks5.py does.

Reply:
I will fix the leaked daemon thread, but prefer not use a process. If the portage somehow killed, the gpg-keepalive daemon process will keep running which is more security sensitive than 0.x second GIL. (It only use 0.152s on my computer). And to implement interprocess communication seems to be more likely to cause problems.

Reply:
I guess the GIL is not much of an issue unless things are cpu bound.